src/backend/distributed/planner/distributed_planner.c

```diff
@@ -965,6 +983,15 @@ CreateDistributedPlan(uint64 planId, bool allowRecursivePlanning, Query *origina
 	{
 		return distributedPlan;
 	}
+	else if (ContainsNullDistKeyTable(originalQuery))
+	{
+		/*
+		 * We only support router queries if the query contains
+		 * reference to a null-dist-key table.
+		 */
+		WrapRouterErrorForNullDistKeyTable(distributedPlan->planningError);
+		RaiseDeferredError(distributedPlan->planningError, ERROR);
```

Comment:
is the plan to remove these lines again once we made a pass on the logical planner?

Reply:
in general making changes to CreateDistributedPlan for a very specific case seems like a slippery slope to me, it seems better to pass it to the logical planner and then error out immediately there

Reply:
> is the plan to remove these lines again once we made a pass on the logical planner?

Yes.

> in general making changes to CreateDistributedPlan for a very specific case seems like a slippery slope, it seems better to pass it to the logical planner and then error out immediately there

Yeah that would be nicer, let me spend some time on that.